tools/GSDumpGUI/Forms/frmMain.cs

```diff
@@ -381,22 +511,60 @@ private void rda_CheckedChanged(object sender, EventArgs e)
                 SelectedRad = Convert.ToInt32(itm.Tag);
         }
 
-        private void txtGSDXDirectory_Leave(object sender, EventArgs e)
+        private void txtGSDXDirectory_Enter(object sender, EventArgs e)
         {
-            Settings.Save();
-            ReloadGsdxDlls();
+            _gsdxPathOld = txtGSDXDirectory.Text;
+        }
 
-            // Auto select GSdx dll
-            _availableGsDlls.Selected = _availableGsDlls.Files.FirstOrDefault();
+        private void txtDumpsDirectory_Enter(object sender, EventArgs e)
+        {
+            _dumpPathOld = txtDumpsDirectory.Text;
+        }
+
+        private void txtGSDXDirectory_Leave(object sender, EventArgs e)
+        {
+            string newpath = txtGSDXDirectory.Text;
+            if (!_gsdxPathOld.ToLower().Equals(newpath.ToLower()))
+                txtGSDXDirectory.Text = _gsdxPathOld;
         }
 
         private void txtDumpsDirectory_Leave(object sender, EventArgs e)
         {
-            Settings.Save();
-            ReloadGsdxDumps();
+            string newpath = txtDumpsDirectory.Text;
+            if(!_dumpPathOld.ToLower().Equals(newpath.ToLower()))
+                txtDumpsDirectory.Text = _dumpPathOld;
+        }
 
-            // Auto select GS dump
-            _availableGsDumps.Selected = _availableGsDumps.Files.FirstOrDefault();
+        private void txtGSDXDirectory_KeyDown(object sender, KeyEventArgs e)
+        {
+            if (e.KeyCode != Keys.Return) return;
```

Comment:
My personal opinion but I don’t like to hide control flow jumps in row 123432543 in a line because it’s harder to read. But yeah there are different opinions about that.